File: day03/part1.swift
var total = 0

while let line = readLine() {
    let expression = #/mul\((\d{1,3}),(\d{1,3})\)/#
    for match in line.matches(of: expression) {
        total += Int(match.output.1)! * Int(match.output.2)! 
    }
}

print(total)

File: day03/part1.js
let total = 0;

for await (const line of console) {
    const regex = new RegExp("mul\\((\\d{1,3}),(\\d{1,3})\\)", "g");
    while (true) {
        const match = regex.exec(line);
        if (!match) break;
        const [wholeMatch, num1, num2] = match;
        total += Number(num1) * Number(num2);
    }
}

console.log(total);